client-py/src/SessionUT.py

```diff
@@ -0,0 +1,482 @@
+    def test_by_str(self):
+        self.session.set_storage_group("root.sg1")
+        self.createTimeSeries(self.session)
+        self.insertByStr(self.session)
+        # SQL tests
+        self.insert_via_sql(self.session)
+        self.query3()
+        # self.insertTabletTest1(self.session, "root.sg1.d1")
+        self.deleteData(self.session)
+        self.query()
+        self.deleteTimeseries(self.session)
+        self.query2()
+        self.insertRecordsByStr(self.session)
+        self.query4()
+
+        # special characters
+        self.session.create_time_series("root.sg1.d1.1_2", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.session.create_time_series("root.sg1.d1.\"1.2.3\"", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.session.create_time_series("root.sg1.d1.\'1.2.4\'", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+
+        self.session.set_storage_group("root.1")
+        self.session.create_time_series("root.1.2.3", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+
+        # Add another storage group to test the deletion of storage group
+        self.session.set_storage_group("root.sg2")
+        self.session.create_time_series("root.sg2.d1.s1", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.deleteStorageGroupTest(self.session)
+
+        # set storage group without creating time series
+        self.session.set_storage_group("root.sg3")
+        self.insertTabletTest1(self.session, "root.sg3.d1")
+
+        # create timeseries without setting storage groups
+        self.session.create_time_series("root.sg4.d1.s1", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.session.create_time_series("root.sg4.d1.s2", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.session.create_time_series("root.sg4.d1.s3", TSDataType.INT64, TSEncoding.RLE, Compressor.SNAPPY)
+        self.insertTabletTest1(self.session, "root.sg4.d1")
+
+        # directly insertion, without any setting of storage groups and time series.
+        self.insertTabletTest1(self.session, "root.sg5.d1")
+
+        # clear
+        self.session.delete_storage_groups(["root.sg3", "root.sg4", "root.sg5", "root.1"])
+
+    def test_session_interfaces_with_disabled_WAL(self):
+        pass
+
+    def query(self):
+        """ JDBC connection required. """
+        pass
+
+    def query2(self):
+        """ JDBC connection required. """
+        pass
+
+    def query3(self):
+        session_data_set = self.session.execute_query_statement("select * from root.sg1.d1")
+        session_data_set.set_fetch_size(1024)
+        count = 0
+        while session_data_set.has_next():
+            index = 1
+            count += 1
+            for f in session_data_set.next().get_fields():
+                self.assertEqual(str(index), f.get_string_value())
+                index += 1
+
+        self.assertEqual(101, count)
+        session_data_set.close_operation_handle()
```

Comment:
It would be better to put the close operation into a finally block, otherwise, one failed test may influence others.